### ISSUE TITLE
LTC not supported in DEX trading ?  (segwit)

### ISSUE THREAD

**Issue by ShorelineCrypto** [webworker01/mmtools#6]
Komodo DeFi Framework (or atomicDEX-api) v2.0.0-beta  is released and Komodo desktop/mobile wallet is under upgrade recently. 

In recent new hardfork style upgrade on wallet and DEX,  LTC now is wallet-only with no trading capability. All trading has to go through LTC-segwit

segwit is not supported in mmtools.  This upgrade make mmtools not usable even with mm2 binary file workaround replacement.

**Comment by ShorelineCrypto** [comment]
smk762: using your repo for a fresh install, then overwrite with 2.0.0-beta mm2 binary to overwrite mmtool old version, then put back my config for LTC.  It fails.  I think the failure was that mmtools as wallet can not enable LTC-segwit, only the legacy LTC wallet is supported. Here are errors:

`

   ./enable ltc-segwit
{
  "error": "rpc:215] dispatcher_legacy:141] lp_commands_legacy:164] lp_coins:4505] mm2 param is not set neither in coins config nor enable request, assuming that coin is not supported"
}
{
  "error": "rpc:215] dispatcher_legacy:141] lp_coins:4060] No such coin LTC-SEGWIT"
}
ubuntu@instance-xx:~/mmtools$ ./enable btc-segwit
{
  "error": "rpc:215] dispatcher_legacy:141] lp_commands_legacy:164] lp_coins:4505] mm2 param is not set neither in coins config nor enable request, assuming that coin is not supported"
}
{
  "error": "rpc:215] dispatcher_legacy:141] lp_coins:4060] No such coin BTC-SEGWIT"
}

~/mmtools$ ./orderbook LTC CHTA
jq: error: segwit/0 is not defined at <top-level>, line 1:
.[$coinkey].price_btc-segwit                      
jq: 1 compile error
jq: error: segwit/0 is not defined at <top-level>, line 1:
.[$coinkey].price_btc-segwit                      
jq: 1 compile error
--Bids (Selling CHTA for LTC)--
Coin    Address                            UUID                                     Price CHTA/LTC           Volume LTC          Volume CHTA       Price LTC/CHTA
----    -------                            ----                                      -------------           ----------           ----------        -------------
jq: error (at <stdin>:1): Cannot iterate over null (null)
                                                                                         0.00000000           0.00000000           0.00000000           0.00000000

--Asks (Buying CHTA with LTC)--
Coin    Address                            UUID                                     Price CHTA/LTC           Volume LTC          Volume CHTA       Price LTC/CHTA
----    -------                            ----                                      -------------           ----------           ----------        -------------
jq: error (at <stdin>:1): Cannot iterate over null (null)
                                                                                         0.00000000           0.00000000           0.00000000           0.00000000

ubuntu@instance-xx:~/atomicDEX-API/target/debug$ ./mm2  --version
AtomicDEX API: 2.0.0-beta_b0fd99e84

`

**Comment by ShorelineCrypto** [comment]
One suggestion from smk for extra testing in discord:

>  smk — Yesterday at 10:57 PM
k, probably best to await @webworker01 fix, I'm not so familiar with all the code, just took a guess for fixing. One other thing you can try is to edit your coins file to set wallet_only to false for LTC
honglu69 — Today at 12:20 AM
Yes, need to wait  @webworker01 fix on LTC pairs trading.  Tried changinge "wallet_only" for LTC from true to false on coins/coin file, the mmtools can place buy an sell order on CHTA/LTC,  but using Desktop new version to trade against that, not normal.    In fact the debug print out in linux says that "LTC wallet only, best order removed".  The trading is not working many times.

**Comment by ShorelineCrypto** [comment]
The errors by using smk's suggestion of making coins file under ~/mmtools/coins/coins  from wallet_only true to false. 
tested both on smk PR branch code, or the default master branch here. Both code failed with different errors. The smk's code error was reported above.  Both ways in mmtools can place orders CHTA/LTC, both show up for desktop GUI to trade against.  smk's branch code sell order was working, desktop was able to buy CHTA with LTC-segwith for 2 sats price. But buy order in mmtools failed, desktop GUI can not sell CHTA for 1 litecoin sats price. 

The original master branch testing again,  sell or buy order of CHTA/LTC pair is not tradable.  In both cases, the Desktop GUI v7.0 wallet generated errors saying "Trade preimage timed out, try again".  

In short, so far LTC is not really working, it can only serve as wallet only under new mm2 v2.0.0-beta upgrade.
